webrtc.html

```diff
@@ -5976,6 +5976,7 @@ <h4>
   readonly attribute DOMString? relatedAddress;
   readonly attribute unsigned short? relatedPort;
   readonly attribute DOMString? usernameFragment;
+  readonly attribute DOMString? relayProtocol;
```

Comment:
How would you define the possible values? https://www.rfc-editor.org/rfc/rfc5928#section-3 is the closest one I could find (Table 1) but that is hardly normative

Reply:
Right now the PR defines the possible values in prose: _"Valid values are "udp", "tcp", or "tls". "_ which seems equivalent to:
```idl
enum RTCIceTransportPolicy {
  "udp",
  "tcp",
  "tls",
};

interface RTCIceCandidate {
  ...
  readonly attribute RTCIceTransportPolicy? relayProtocol;
};
```
If those are the only valid values then we generally prefer express things in WebIDL when possible.

If those are not the only values then we should rephrase.

Reply:
> If those are the only valid values then we generally prefer expressing things in WebIDL when possible.

Or at least that's been my impression. These are only ever used as output, not input so I guess it doesn't matter a whole lot (do we get any free idlharness wpt tests)?  I see the stats spec is just using DOMString, and I couldn't find any design principles on this. The WebIDL spec showed no examples of a `readonly attribute` using `enum`... YMMV

Reply:
Yes but for the other enums we have normative references from which we can draw the values (e.g. https://w3c.github.io/webrtc-pc/#rtciceprotocol-enum but note that while two of the values are the same the semantics is very different)

For stats this is actually a TODO hidden here: https://github.com/w3c/webrtc-stats/blob/main/webrtc-stats.html#L2964

Reply:
[RTCIceTcpCandidateType](https://w3c.github.io/webrtc-pc/#dom-rtcicetcpcandidatetype) is sort of similar here.
Having WebIDL enum is a small UA developer improvement as it can automate some binding generated code.

Reply:
RTCIceServerTransportProtocol won against RTCIceTurnRelayProtocol and variants thereof.